src/userflow.ts

```diff
@@ -206,6 +207,37 @@ interface Deferred {
 // empty object instead of `window`.
 var w: WindowWithUserflow = typeof window === 'undefined' ? ({} as any) : window
 var userflow = w.userflow
+const history = w.history
+
+// indicates if the history API's pushState and replaceState are patched with custom event emitters
+w.__userflowStatePatched = true
+
+// patch the history API's
+// pushState method to emit userflow:pushstate and
+// replaceState method to emit userflow:replacestate
+const originalPushState = history.pushState
+const originalReplaceState = history.replaceState
+
+function overrideHistoryMethods(
+  method: (...args: any[]) => void,
+  eventName: string
+) {
+  return function (...args: any[]) {
+    const event = new CustomEvent(eventName)
+    const ret = method.call(history, ...args)
```

Comment:
Dont use spread operator and introduce a new babel dependency. Instead use 
```suggestion
    const ret = method.apply(history, arguments)
```